File: TestConnection.java
import java.sql.Connection;
import java.sql.DriverManager;

public class TestConnection {
    public static void main(String[] args) {
        String[] usernames = {"Thanhjash", "thanhjash", "postgres"};
        String password = "ViLink_17";
        String url = "jdbc:postgresql://aws-0-ap-southeast-1.pooler.supabase.com:6543/postgres?sslmode=require";

        for (String username : usernames) {
            System.out.println("\nTesting with username: " + username);
            try (Connection conn = DriverManager.getConnection(url, username, password)) {
                System.out.println("✅ SUCCESS! Connected with username: " + username);
                System.out.println("Connection valid: " + conn.isValid(5));
                break;
            } catch (Exception e) {
                System.out.println("❌ FAILED with " + username + ": " + e.getMessage());
            }
        }
    }
}

File: testConnection.js
const { Client } = require('pg');

const credentials = [
  { user: 'postgres.doxksbweeaxtewrlcvat', password: 'QbOaG8jWXkphbZnQ' },
  { user: 'postgres', password: 'QbOaG8jWXkphbZnQ' },
  { user: 'Thanhjash', password: 'QbOaG8jWXkphbZnQ' },
  { user: 'thanhjash', password: 'QbOaG8jWXkphbZnQ' },
];

const hosts = [
  { host: 'aws-0-ap-southeast-1.pooler.supabase.com', port: 6543, name: 'Session Pooler' },
  { host: 'db.doxksbweeaxtewrlcvat.supabase.co', port: 5432, name: 'Direct Connection' },
];

async function testConnection() {
  for (const hostInfo of hosts) {
    console.log(`\n========== Testing: ${hostInfo.name} ==========`);
    console.log(`Host: ${hostInfo.host}:${hostInfo.port}`);

    for (const cred of credentials) {
      console.log(`\n  Testing username: ${cred.user}`);

      const client = new Client({
        host: hostInfo.host,
        port: hostInfo.port,
        database: 'postgres',
        user: cred.user,
        password: cred.password,
        ssl: { rejectUnauthorized: false }
      });

      try {
        await client.connect();
        console.log(`  ✅ SUCCESS! Connected with username: ${cred.user}`);
        console.log(`\n  🎉 WORKING CREDENTIALS:`);
        console.log(`  Host: ${hostInfo.host}`);
        console.log(`  Port: ${hostInfo.port}`);
        console.log(`  User: ${cred.user}`);
        console.log(`  Password: ${cred.password}`);

        await client.end();
        process.exit(0);
      } catch (err) {
        console.log(`  ❌ FAILED: ${err.message}`);
      }
    }
  }

  console.log('\n❌ All combinations failed!');
  process.exit(1);
}

testConnection();
